presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -851,15 +849,22 @@ public void commitInsert(ConnectorSession session, ConnectorInsertTableHandle in
         }
     }
 
-    private static Partition createPartition(Table table, PartitionUpdate partitionUpdate)
+    private Partition createPartition(Table table, PartitionUpdate partitionUpdate)
     {
         List<String> values = HivePartitionManager.extractPartitionKeyValues(partitionUpdate.getName());
         Partition partition = new Partition();
         partition.setDbName(table.getDbName());
         partition.setTableName(table.getTableName());
         partition.setValues(values);
-        partition.setSd(table.getSd().deepCopy());
-        partition.getSd().setLocation(partitionUpdate.getTargetPath());
+
+        if (respectTableFormat) {
+            partition.setSd(table.getSd().deepCopy());
+            partition.getSd().setLocation(partitionUpdate.getTargetPath());
+        }
+        else {
+            partition.setSd(makeStorageDescriptor(table.getDbName(), defaultStorageFormat, new Path(partitionUpdate.getTargetPath()), table.getSd().getCols()));
```

Comment:
It looks like you're using the wrong value here. You pass in `table.getDbName()`, but it's expecting `tableName` which is what you use in the other call to `makeStorageDescriptor`